### PR TITLE
fix(http): Adjust hostname display for Windows when using 0.0.0.0

### LINKED ISSUE #24687: `deno serve` prints that it listens on localhost instead of 0.0.0.0
Version: Deno 1.45.3

```ts
// server.ts
export default {
  fetch(_request: Request) {
    return new Response("Hello, world!");
  },
};
```

```sh
$ deno serve --host 0.0.0.0 server.ts
deno serve: Listening on http://localhost:8000/
```

### PR DESCRIPTION
**Pull request by yazan-abdalrahman**
fix #24687

### CODE REVIEW

File: ext/http/00_serve.ts [hunk 1]
```diff
@@ -657,7 +657,8 @@ function serve(arg1, arg2) {
   // If the hostname is "0.0.0.0", we display "localhost" in console
   // because browsers in Windows don't resolve "0.0.0.0".
   // See the discussion in https://github.com/denoland/deno_std/issues/1165
-  const hostname = addr.hostname == "0.0.0.0" || addr.hostname == "::"
+  const hostname = (addr.hostname == "0.0.0.0" || addr.hostname == "::") &&
+      (Deno.build.os !== "windows")
```
Comment (bartlomieju) [inline ext/http/00_serve.ts:661]: This should be `Deno.build.os === "windows"`. We only want to change it to `localhost` for Windows as per the comment about

Comment (yazan-abdalrahman) [reply ext/http/00_serve.ts]: done

File: tests/unit/serve_test.ts [hunk 2]
```diff
@@ -741,7 +741,7 @@ Deno.test({ permissions: { net: true } }, async function httpServerPort0() {
 });
 
 Deno.test(
-  { permissions: { net: true } },
+  { permissions: { net: true }, ignore: Deno.build.os !== "windows" },
```
Comment (kt3k) [inline tests/unit/serve_test.ts:744]: I think we should change the regexp below to match both patterns instead of ignoring the test case in non-windows.

**Comment by kt3k** [comment]
Modified the test cases. Now we don't ignore test cases on both platforms.

@bartlomieju PTAL

**Review by bartlomieju — APPROVED** [review]
LGTM too!

**Pull request merged**